spec/board_spec.rb

```diff
@@ -41,4 +41,41 @@
       expect(board.valid_coordinate?("A5")).to be false
     end
   end
+
+    
+  describe "#valid_placement?" do
+    it "returns false if the number of coordinates does not match the ship's length" do
+      board = Board.new
+      cruiser = Ship.new("Cruiser", 3)
+      submarine = Ship.new("Submarine", 2)
+      expect(board.valid_placement?(cruiser, ["A1", "A2"])).to eq(false)
+      expect(board.valid_placement?(submarine, ["A2", "A3", "A4"])).to eq(false)
+    end
+
+    it "returns false if the coordinates are not consecutive" do
+      board = Board.new
+      cruiser = Ship.new("Cruiser", 3)
+      submarine = Ship.new("Submarine", 2)
+      expect(board.valid_placement?(cruiser, ["A1", "A2", "A4"])).to eq(false)
+      expect(board.valid_placement?(submarine, ["A1", "C1"])).to eq(false)
+      expect(board.valid_placement?(cruiser, ["A3", "A2", "A1"])).to eq(false)
```

Comment:
why is this placement false even though A3,A2,A1 are consecutive?

Reply:
This is a great catch Kevin! It seems I forgot to write edge casing in the case the values we pass in are not in order! I'll get right on that!

Reply:
Glad I was able to help with the code even in a small way! The code looks great ! Awesome work!